lib/matplotlib/backends/backend_tkagg.py

```diff
@@ -731,11 +730,10 @@ def set_message(self, s):
 
     def draw_rubberband(self, event, x0, y0, x1, y1):
         height = self.canvas.figure.bbox.height
-        y0 =  height-y0
-        y1 =  height-y1
-        try: self.lastrect
-        except AttributeError: pass
-        else: self.canvas._tkcanvas.delete(self.lastrect)
+        y0 = height - y0
+        y1 = height - y1
+        if hasattr(self, "lastrect"):
```

Comment:
The following patch
```
diff --git a/lib/matplotlib/backends/backend_tkagg.py b/lib/matplotlib/backends/backend_tkagg.py
index af0ed62f9..4d212efe8 100644
--- a/lib/matplotlib/backends/backend_tkagg.py
+++ b/lib/matplotlib/backends/backend_tkagg.py
@@ -736,7 +736,14 @@ class NavigationToolbar2TkAgg(NavigationToolbar2, Tk.Frame):
         try: self.lastrect
         except AttributeError: pass
         else: self.canvas._tkcanvas.delete(self.lastrect)
+        from time import perf_counter
+        start = perf_counter()
+        elapsed_dummy = perf_counter() - start
+        start = perf_counter()
         self.lastrect = self.canvas._tkcanvas.create_rectangle(x0, y0, x1, y1)
+        elapsed_draw = perf_counter() - start
+        elapsed_corrected = elapsed_draw - elapsed_dummy
+        print(f"create_rectangle took {elapsed_corrected * 1e6}us")
```
shows that the call to create_rectangle that's just below takes ~100-300us, so I think worrying about 50ns for the sake of clarity is totally overkill.